builtin/logical/transit/path_config_test.go

```diff
@@ -6,15 +6,264 @@ import (
+	archive, err = p.LoadArchive(namespace.RootContext(nil), storage)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(archive.Keys) != 3 {
+		t.Fatalf("bad: len of archived keys; expected: 3, actual: %d", len(archive.Keys))
+	}
+
+	// Min decryption version should not be less than trimmed min version
+	req.Data = map[string]interface{}{
+		"min_decryption_version": 1,
+	}
+	doErrReq(t, req)
+
+	// Min encryption version should not be less than trimmed min version
+	req.Data = map[string]interface{}{
+		"min_encryption_version": 2,
+	}
+	doErrReq(t, req)
+
+	// Rotate 5 more times
+	for i := 0; i < 5; i++ {
+		doReq(t, &logical.Request{
+			Path:      "keys/aes/rotate",
+			Storage:   storage,
+			Operation: logical.UpdateOperation,
+		})
+	}
+
+	// Archive: 3, 4, 5, 6, 7, 8, 9, 10
+	archive, err = p.LoadArchive(namespace.RootContext(nil), storage)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(archive.Keys) != 8 {
+		t.Fatalf("bad: len of archived keys; expected: 8, actual: %d", len(archive.Keys))
+	}
+
+	// Set min encryption version to 7
+	req.Path = "keys/aes/config"
+	req.Data = map[string]interface{}{
+		"min_encryption_version": 7,
+	}
+	doReq(t, req)
+
+	// Set min decryption version to 7
+	req.Data = map[string]interface{}{
+		"min_decryption_version": 7,
+	}
+	doReq(t, req)
+
+	// Trimmed all versions before 7
+	req.Data = map[string]interface{}{
+		"trimmed_min_version": 7,
+	}
+	doReq(t, req)
+
+	// Archive: 7, 8, 9, 10
+	archive, err = p.LoadArchive(namespace.RootContext(nil), storage)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(archive.Keys) != 4 {
+		t.Fatalf("bad: len of archived keys; expected: 4, actual: %d", len(archive.Keys))
+	}
+
+	// Read the key
+	req.Path = "keys/aes"
+	req.Operation = logical.ReadOperation
+	resp := doReq(t, req)
+	keys := resp.Data["keys"].(map[string]int64)
+	if len(keys) != 4 {
+		t.Fatalf("bad: number of keys; expected: 4, actual: %d", len(keys))
+	}
+
+	// Test if moving the min_encryption_version and min_decryption_versions
+	// are working fine
+
+	// Set min encryption version to 10
+	req.Path = "keys/aes/config"
+	req.Operation = logical.UpdateOperation
+	req.Data = map[string]interface{}{
+		"min_encryption_version": 10,
+	}
+	doReq(t, req)
+	if p.MinEncryptionVersion != 10 {
+		t.Fatalf("failed to set min encryption version")
+	}
+
+	// Set min decryption version to 9
+	req.Data = map[string]interface{}{
+		"min_decryption_version": 9,
+	}
+	doReq(t, req)
+	if p.MinDecryptionVersion != 9 {
+		t.Fatalf("failed to set min encryption version")
+	}
+
+	// Reduce the min decryption version to 8
+	req.Data = map[string]interface{}{
+		"min_decryption_version": 8,
+	}
+	doReq(t, req)
+	if p.MinDecryptionVersion != 8 {
+		t.Fatalf("failed to set min encryption version")
+	}
+
+	// Reduce the min encryption version to 8
+	req.Data = map[string]interface{}{
+		"min_encryption_version": 8,
+	}
+	doReq(t, req)
+	if p.MinDecryptionVersion != 8 {
+		t.Fatalf("failed to set min decryption version")
+	}
```

Comment:
Would it be possible to restructure this test as a table test? It seems there are several tests scenarios above testing the same process, but not knowing the code well enough maybe it's testing a series of steps and compose a larger scenario?

Reply:
I am not sure if this test can be restructured. Unless I am understanding it wrong, the table tests are a good fit for smaller scenarios where-in each test will do a discrete number of steps. This test is staging the policy in such a way that it is getting modified continuously and the steps are relying on the current state of the policy/archive to verify the behavior. So, yes it is composing a relatively larger scenario and for the very reason, it is not suitable as a table test. Does this make sense to you?

Reply:
>This test is staging the policy in such a way that it is getting modified continuously and the steps are relying on the current state of the policy/archive to verify the behavior.

Makes total sense, I couldn't tell if this was the case or not, being unfamiliar. Thanks for explaining 👍